### PR TITLE
Add semeval 2020 task 11

### PR DESCRIPTION
**Pull request by ZacharySBrown**
Adding in propaganda detection task (task 11) from Sem Eval 2020

### CODE REVIEW

File: datasets/sem_eval_2020_task_11/sem_eval_2020_task_11.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        if labels:
+            si_labels_dir = os.path.join(data_dir, f"{split}-labels-task-si")
+            tc_labels_dir = os.path.join(data_dir, f"{split}-labels-task-flc-tc")
+
+        # Iterate through each article
+        for id_, key in enumerate(keys):
+            text_path = os.path.join(articles_path, key + ".txt")
+
+            # Read the text for the article
+            text = open(text_path, encoding="utf-8").read()
```
Comment (SBrandeis) [inline datasets/sem_eval_2020_task_11/sem_eval_2020_task_11.py:170]: Won't this `open` statement without a call to `.close()` leak the file descriptor ?
I'm more used to this syntax:
```
with open(path, mode) as fd:
    [use fd]
```
Which makes sure `.close()` is called on the file descriptor when exiting the context manager.

File: datasets/sem_eval_2020_task_11/README.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+- **Leaderboard:** [PTC Tasks Leaderboard](https://propaganda.qcri.org/ptc/leaderboard.php)
+- **Point of Contact:** [Task organizers contact](semeval-2020-task-11-organizers@googlegroups.com)
+
+### Dataset Summary
+
+Propagandistic news articles use specific techniques to convey their message, such as whataboutism, red Herring, and name calling, among many others. The Propaganda Techniques Corpus (PTC) allows to study automatic algorithms to detect them. We provide a permanent leaderboard to allow researchers both to advertise their progress and to be up-to-speed with the state of the art on the tasks offered (see below for a definition).
+
+### Supported Tasks and Leaderboards
+
+For each of the tasks tagged for this dataset, give a brief description of the tag, metrics, and suggested models (with a link to their HuggingFace implementation if available). Give a similar description of tasks that were not covered by the structured tag set (repace the `task-category-tag` with an appropriate `other:other-task-name`).
```
Comment (SBrandeis) [inline datasets/sem_eval_2020_task_11/README.md:60]: The placeholder can be removed
```suggestion
```

File: datasets/sem_eval_2020_task_11/README.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    "technique": [7,8, ...]
+  },
+  "text": "Newt Gingrich: The truth about Trump, Putin, and Obama\n\nPresident Trump..."
+}
+
+```
+
+### Data Fields
+
+List and describe the fields present in the dataset. Mention their data type, and whether they are used as input or output in any of the tasks the dataset currently supports. If the data has span indices, describe their attributes, such as whether they are at the character level or word level, whether they are contiguous or not, etc. If the datasets contains example IDs, state whether they have an inherent meaning, such as a mapping to other datasets or pointing to relationships between data points.
```
Comment (SBrandeis) [inline datasets/sem_eval_2020_task_11/README.md:94]:
```suggestion
```

File: datasets/sem_eval_2020_task_11/README.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  - `start_char_offset`: The start character offset of the span for the SI task
+  - `end_char_offset`: The end character offset of the span for the SI task
+- `technique_classification`: a dictionary feature containing:
+  - `start_char_offset`: The start character offset of the span for the TC task
+  - `end_char_offset`: The start character offset of the span for the TC task
+  - `technique`: the propaganda technique classification label, with possible values including `Appeal_to_Authority`, `Appeal_to_fear-prejudice`, `Bandwagon,Reductio_ad_hitlerum`, `Black-and-White_Fallacy`, `Causal_Oversimplification`.
+
+### Data Splits
+
+Describe and name the splits in the dataset if there are more than one.
```
Comment (SBrandeis) [inline datasets/sem_eval_2020_task_11/README.md:107]:
```suggestion
```

File: datasets/sem_eval_2020_task_11/README.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+Describe any criteria for splitting the data, if used. If their are differences between the splits (e.g. if the training annotations are machine-generated and the dev and test ones are created by humans, or if different numbers of annotators contributed to each example), describe them here.
```
Comment (SBrandeis) [inline datasets/sem_eval_2020_task_11/README.md:109]:
```suggestion
```

File: datasets/sem_eval_2020_task_11/README.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+Provide the sizes of each split. As appropriate, provide any descriptive statistics for the features, such as average length.  For example:
```
Comment (SBrandeis) [inline datasets/sem_eval_2020_task_11/README.md:111]:
```suggestion
```

**Review by SBrandeis — COMMENTED** [review]
Awesome contribution, thanks a lot !

Before merging, I would recommend removing the default values in the dataset card where information is provided. I also wonder if NER is the correct task id tag for this dataset. @lhoestq WDYT ?

I also have a question regarding the syntax you use to `open` files.

File: datasets/sem_eval_2020_task_11/sem_eval_2020_task_11.py (hunk 1 above)
Comment (ZacharySBrown) [reply datasets/sem_eval_2020_task_11/sem_eval_2020_task_11.py:170]: Oh, good call, happy to change this to use the syntax above!

**Comment by ZacharySBrown** [comment]
@SBrandeis : Thanks for the feedback! Just updated to use context manager for the `open`s and removed the placeholder text from the `README`!

**Comment by SBrandeis** [comment]
Great, thanks @ZacharySBrown !
Failing tests seem to be unrelated to your changes, merging the current master branch into yours should fix them.

File: datasets/sem_eval_2020_task_11/README.md [hunk 7]
```diff
@@ -0,0 +1,198 @@
+---
+annotations_creators:
+- expert-generated
+language_creators: []
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/README.md:4]:
```suggestion
language_creators:
- original
```

File: datasets/sem_eval_2020_task_11/README.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+The annotation job consisted of both spotting a propaganda snippet and, at the same time, labeling
+it with a specific propaganda technique. The annotation guidelines are shown in the appendix; they
+are also available online.4 We ran the annotation in two phases: (i) two annotators label an article
+independently and (ii) the same two annotators gather together with a consolidator to discuss dubious
+instances (e.g., spotted only by one annotator, boundary discrepancies, label mismatch, etc.). This protocol
+was designed after a pilot annotation stage, in which a relatively large number of snippets had been spotted
+by one annotator only. The annotation team consisted of six professional annotators from A Data Pro trained to spot and label the propaganda snippets from free text. The job was carried out on an instance of
+the Anafora annotation platform (Chen and Styler, 2013), which we tailored for our propaganda annotation
+task. Figure 2 shows an example of an article and its annotations.
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/README.md:144]:
```suggestion
task.
```

File: datasets/sem_eval_2020_task_11/sem_eval_2020_task_11.py [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+        # Get the main path for the articles
+        articles_path = os.path.join(data_dir, f"{split}-articles")
+
+        # If the split has labels, get the main paths for labels
+        if labels:
+            si_labels_dir = os.path.join(data_dir, f"{split}-labels-task-si")
+            tc_labels_dir = os.path.join(data_dir, f"{split}-labels-task-flc-tc")
+
+        # Iterate through each article
+        for id_, key in enumerate(keys):
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/sem_eval_2020_task_11.py:166]:
```suggestion
        for id_, key in enumerate(sorted(keys)):
```
we need the order to be deterministic here

File: datasets/sem_eval_2020_task_11/README.md [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+### Supported Tasks and Leaderboards
+
+For each of the tasks tagged for this dataset, give a brief description of the tag, metrics, and suggested models (with a link to their HuggingFace implementation if available). Give a similar description of tasks that were not covered by the structured tag set (repace the `task-category-tag` with an appropriate `other:other-task-name`).
+
+- `named-entity-recognition`: The dataset can be used to train a model for token classifcation for both a span identification task as well as a technique classification task. Success on this task is typically measured by achieving a high (micro-averaged)[f1](https://huggingface.co/metrics/f1) for the technique classification task. ([Roberta](https://huggingface.co/model_name) with an additional CRF layer currently achieves the a micro-averaged f1 of 0.637 on the TC task. This task has an active leaderboard which can be found [here](https://propaganda.qcri.org/ptc/leaderboard.php).
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/README.md:62]:
```suggestion
- `named-entity-recognition`: The dataset can be used to train a model for token classifcation for both a span identification task as well as a technique classification task. Success on this task is typically measured by achieving a high (micro-averaged)[f1](https://huggingface.co/metrics/f1) for the technique classification task. ([Roberta](https://huggingface.co/roberta-base) with an additional CRF layer currently achieves the a micro-averaged f1 of 0.637 on the TC task. This task has an active leaderboard which can be found [here](https://propaganda.qcri.org/ptc/leaderboard.php). 
```

File: datasets/sem_eval_2020_task_11/README.md [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+- **Leaderboard:** [PTC Tasks Leaderboard](https://propaganda.qcri.org/ptc/leaderboard.php)
+- **Point of Contact:** [Task organizers contact](semeval-2020-task-11-organizers@googlegroups.com)
+
+### Dataset Summary
+
+Propagandistic news articles use specific techniques to convey their message, such as whataboutism, red Herring, and name calling, among many others. The Propaganda Techniques Corpus (PTC) allows to study automatic algorithms to detect them. We provide a permanent leaderboard to allow researchers both to advertise their progress and to be up-to-speed with the state of the art on the tasks offered (see below for a definition).
+
+### Supported Tasks and Leaderboards
+
+For each of the tasks tagged for this dataset, give a brief description of the tag, metrics, and suggested models (with a link to their HuggingFace implementation if available). Give a similar description of tasks that were not covered by the structured tag set (repace the `task-category-tag` with an appropriate `other:other-task-name`).
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/README.md:60]:
```suggestion
```

File: datasets/sem_eval_2020_task_11/README.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+multilinguality:
+- monolingual
+size_categories:
+- n<1K
+source_datasets:
+- original
+task_categories:
+- structure-prediction
+task_ids:
+- named-entity-recognition
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/README.md:18]:
```suggestion
task_categories:
task_categories:
- text-classification
- structure-prediction
task_ids:
- text-classification-other-language-identification
- structure-prediction-other-propaganda-span-identi
```

File: datasets/sem_eval_2020_task_11/sem_eval_2020_task_11.py [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+advertise their progress and to be up-to-speed with the state of the art on the
+tasks offered (see below for a definition).
+"""
+
+_HOMEPAGE = "https://propaganda.qcri.org/ptc/index.html"
+
+_LICENSE = ""
+
+
+class SemEval2020Task11(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/sem_eval_2020_task_11.py:49]:
```suggestion
class SemEval_2020Task_11(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `sem_eval_2020_task_11`

File: datasets/sem_eval_2020_task_11/dataset_infos.json [hunk 14]
```diff
@@ -0,0 +1 @@
...t64", "id": null, "_type": "Value"}, "technique": {"num_classes": 14, "names": ["Appeal_to_Authority", "Appeal_to_fear-prejudice", "Bandwagon,Reductio_ad_hitlerum", "Black-and-White_Fallacy", "Causal_Oversimplification", "Doubt", "Exaggeration,Minimisation", "Flag-Waving", "Loaded_Language", "Name_Calling,Labeling", "Repetition", "Slogans", "Thought-terminating_Cliches", "Whataboutism,Straw_Men,Red_Herring"], "names_file": null, "id": null, "_type": "ClassLabel"}}, "length": -1, "id": null, "_type": "Sequence"}}, "post_processed": null, "supervised_keys": null, "builder_name": "sem_eval2020_task11", "config_name": "default", "version": {"version_str": "1.1.0", "description": null, "major": 1, "minor": 1, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 2351189, "num_examples": 371, "dataset_name": "sem_eval2020_task11"}, "test": {"name": "test", "num_bytes": 410104, "num_examples": 90, "dataset_name": "sem_eval2020_task11"}, "validation": {"name": "validation", "num_bytes": 394906, "num_examples": 75, "dataset_name": "sem_eval2020_task11"}}, "download_checksums": {}, "download_size": 0, "post_processing_size": null, "dataset_size": 3156199, "size_in_bytes": 3156199}}
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/dataset_infos.json:1]:
```suggestion
{"default": {"description": "Propagandistic news articles use specific techniques to convey their message,\nsuch as whataboutism, red Herring, and name calling, among many others.\nThe Propaganda Techniques Corpus (PTC) allows to study automatic algorithms to\ndetect them. We provide a permanent leaderboard to allow researchers both to\nadvertise their progress and to be up-to-speed with the state of the art on the\ntasks offered (see below for a definition).\n", "citation": "\n@misc{martino2020semeval2020,\n      title={SemEval-2020 Task 11: Detection of Propaganda Techniques in News Articles},\n      author={G. Da San Martino and A. Barr\u00f3n-Cede\u00f1o and H. Wachsmuth and R. Petrov and P. Nakov},\n      year={2020},\n      eprint={2009.02696},\n      archivePrefix={arXiv},\n      primaryClass={cs.CL}\n}\n", "homepage": "https://propaganda.qcri.org/ptc/index.html", "license": "", "features": {"text": {"dtype": "string", "id": null, "_type": "Value"}, "span_identification": {"feature": {"start_char_offset": {"dtype": "int64", "id": null, "_type": "Value"}, "end_char_offset": {"dtype": "int64", "id": null, "_type": "Value"}}, "length": -1, "id": null, "_type": "Sequence"}, "technique_classification": {"feature": {"start_char_offset": {"dtype": "int64", "id": null, "_type": "Value"}, "end_char_offset": {"dtype": "int64", "id": null, "_type": "Value"}, "technique": {"num_classes": 14, "names": ["Appeal_to_Authority", "Appeal_to_fear-prejudice", "Bandwagon,Reductio_ad_hitlerum", "Black-and-White_Fallacy", "Causal_Oversimplification", "Doubt", "Exaggeration,Minimisation", "Flag-Waving", "Loaded_Language", "Name_Calling,Labeling", "Repetition", "Slogans", "Thought-terminating_Cliches", "Whataboutism,Straw_Men,Red_Herring"], "names_file": null, "id": null, "_type": "ClassLabel"}}, "length": -1, "id": null, "_type": "Sequence"}}, "post_processed": null, "supervised_keys": null, "builder_name": "sem_eval_2020_task_11", "config_name": "default", "version": {"version_str": "1.1.0", "description": null, "major": 1, "minor": 1, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 2351189, "num_examples": 371, "dataset_name": "sem_eval_2020_task_11"}, "test": {"name": "test", "num_bytes": 410104, "num_examples": 90, "dataset_name": "sem_eval_2020_task_11"}, "validation": {"name": "validation", "num_bytes": 394906, "num_examples": 75, "dataset_name": "sem_eval_2020_task_11"}}, "download_checksums": {}, "download_size": 0, "post_processing_size": null, "dataset_size": 3156199, "size_in_bytes": 3156199}}
```

`sem_eval2020_task11`-> `sem_eval_2020_task_11`

**Review by lhoestq — APPROVED** [review]
Awesome thanks !

File: datasets/sem_eval_2020_task_11/README.md [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+- **Point of Contact:** [Task organizers contact](semeval-2020-task-11-organizers@googlegroups.com)
+
+### Dataset Summary
+
+Propagandistic news articles use specific techniques to convey their message, such as whataboutism, red Herring, and name calling, among many others. The Propaganda Techniques Corpus (PTC) allows to study automatic algorithms to detect them. We provide a permanent leaderboard to allow researchers both to advertise their progress and to be up-to-speed with the state of the art on the tasks offered (see below for a definition).
+
+### Supported Tasks and Leaderboards
+
+
+- `named-entity-recognition`: The dataset can be used to train a model for token classifcation for both a span identification task as well as a technique classification task. Success on this task is typically measured by achieving a high (micro-averaged)[f1](https://huggingface.co/metrics/f1) for the technique classification task. ([Roberta](https://huggingface.co/model_name) with an additional CRF layer currently achieves the a micro-averaged f1 of 0.637 on the TC task. This task has an active leaderboard which can be found [here](https://propaganda.qcri.org/ptc/leaderboard.php).
```
Comment (lhoestq) [inline datasets/sem_eval_2020_task_11/README.md:65]:
```suggestion
```

**Pull request merged**